app/storyedit/models/story.model.ts

```diff
@@ -0,0 +1,125 @@
+import {Observable} from 'rxjs';
+import {CmsService} from '../../shared/cms/cms.service';
+import {HalDoc} from '../../shared/cms/haldoc';
+
+class StoryPayload {
+  id: number = null;
+  title: string = '';
+  shortDescription: string = '';
+  modifiedAt: Date = null;
+  toJSON(): {} {
+    return {
+      title: this.title,
+      shortDescription: this.shortDescription
+    };
+  }
+}
+
+class StoryChanged {
+  title: boolean = false;
+  shortDescription: boolean = false;
+}
+
+class StoryInvalid {
+  title: string;
+  shortDescription: string;
+}
+
+export class StoryModel {
```

Comment:
Yeah, I'm actually fairly unhappy with this model class.  I went through way too much effort to run validations "on field set" rather than "on demand".  Currently refactoring it as I work on putting in tags, into a more normal `isValid('title')` sort of thing.